xray_vision/backend/mpl/cross_section_2d.py

```diff
@@ -539,9 +626,20 @@ def format_coord(x, y):
             if col >= 0 and col < numcols and row >= 0 and row < numrows:
                 # if it does, grab the value
                 z = self._imdata[row, col]
-                return "X: {x:d} Y: {y:d} I: {i:.2f}".format(x=col, y=row, i=z)
+                if self.extent_labels is not None:
+                    return "X: {x:g} Y: {y:g} I: {i:g}".format(
+                    x=col / im_shape[1] * (self.extent_labels[1] -
```

Comment:
I think if you set the image extent you won't need this anymore
you can do this in (look for this line in code)
```
self._init_artists(image, extent=extent)
```
and update things accordingly to get it working.